pkg/metrics/metrics.go

```diff
@@ -0,0 +1,19 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	ReconcileError = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "observability_operator_reconcile_error_total",
```

Comment:
I think we should bé able to create the namespace in a more central place (not sure how but I know we can) so we could define the prefix to Always bé observability_operator but that's okay for now.
I would however make this a custom metrics like monitoring_mimir_head_series_errors_count or smth. WDYT?

Reply:
Also, it probably should bé under thé monitoring Namespace ;)
I don't really liké the generic réconciliation errors because we already exposé those metrics if a réconciliation loops returns an error

Reply:
This metric is very specific to this operator I would prefer to keep is named after the operator.

controller runtime do expose `controller_runtime_reconcile_errors_total` but AFAIK there are no other reconciliation error metric exposed here.

Reply:
I was thinking if smth like observability_operator_monitoring_mimir_head_series_errors_count 😂

Reply:
But m'y point is mostly that I think we should have a spécialized metric for this error instead of a generic réconciliation error metric

Reply:
How about `observability_operator_mimir_head_series_query_errors_total` ? specific enought ? :D

Reply:
That lo<3 this is just perfect for me :)